lib/feature/home_page/widgets/pokemon_search_page.dart

```diff
@@ -0,0 +1,44 @@
+import 'package:flutter/material.dart';
+import 'package:pokedex_async_redux/api/models/pokemon_model.dart';
+import 'package:pokedex_async_redux/feature/home_page/widgets/pokemon_tile.dart';
+
+class PokemonSearchPage extends SearchDelegate {
+  PokemonSearchPage({required this.pokemonList});
+  final List<Pokemon> pokemonList;
+  @override
+  List<Widget>? buildActions(BuildContext context) => [
+        IconButton(
+          icon: Icon(Icons.clear),
+          onPressed: () {
+            if (query.isEmpty) {
+              close(context, null);
+            } else {
+              query = '';
+            }
+          },
+        ),
+      ];
+
+  @override
+  Widget? buildLeading(BuildContext context) => null;
+
+  @override
+  Widget buildResults(BuildContext context) => Padding(
+        padding: EdgeInsets.fromLTRB(10, 10, 10, 0),
+        child: GridView.count(
+            crossAxisCount: 2,
+            mainAxisSpacing: 5,
+            crossAxisSpacing: 5,
+            childAspectRatio: 3 / 2,
+            children: pokemonList
+                .where((element) => element.name!.contains(query))
```

Comment:
use filter.

Reply:
should it really be contain? because if I type `a` all names that contains an a will be shown. Think of how search is implemented when you're trying to find a name in your contacts